responses/tests/test_responses.py

```diff
@@ -2035,6 +2036,88 @@ def run():
     assert_reset()
 
 
+def test_response_and_requests_mock_calls_are_equal():
+    @responses.activate
+    def run():
+        rsp = responses.add(responses.GET, "http://www.example.com")
+        rsp2 = responses.add(responses.GET, "http://www.example.com/1")
+
+        requests.get("http://www.example.com")
+        requests.get("http://www.example.com/1")
+
+        assert len(responses.calls) == 2
+        assert rsp.call_count == 1
+        assert rsp.calls[0] is responses.calls[0]
+        assert rsp2.call_count == 1
+        assert rsp2.calls[0] is responses.calls[1]
+
+    run()
+    assert_reset()
+
+
+def test_response_call_request():
+    @responses.activate
+    def run():
+        rsp = responses.add(responses.GET, "http://www.example.com")
+        rsp2 = responses.add(
+            responses.PUT, "http://www.foo.bar/42/", json={"id": 42, "name": "Bazz"}
+        )
+
+        requests.get("http://www.example.com")
+        requests.get("http://www.example.com?hello=world")
+        requests.put(
+            "http://www.foo.bar/42/",
+            json={"name": "Bazz"},
+        )
+
+        assert rsp.call_count == 2
+        request = rsp.calls[0].request
+        assert request.url == "http://www.example.com/"
+        assert request.method == "GET"
+        request = rsp.calls[1].request
```

Comment:
I think we can make it a bit more readable by enumerating `request` objects as well
```suggestion
        assert rsp.call_count == 2
        
        request1 = rsp.calls[0].request
        assert request.url == "http://www.example.com/"
        assert request.method == "GET"
        
        request2 = rsp.calls[1].request
```

and so on